http3/server.go

```diff
@@ -205,6 +259,85 @@ func (s *Server) handleConn(sess quic.EarlySession) {
 	}
 }
 
+func (s *Server) addCtrlStream(c *ctrlStream) {
+	s.mutex.Lock()
+	if s.ctrlStreams == nil {
+		s.ctrlStreams = make(map[*ctrlStream]struct{})
+	}
+	s.ctrlStreams[c] = struct{}{}
```

Comment:
I don't think you need any map here.

Reply:
Yes I have removed it. Now using a channel to signal to control streams